virt_lightning/shell.py

```diff
@@ -20,47 +22,75 @@
-    print("Starting:")
-    status_line = ""
-    for host in host_definitions:
-        if "name" not in host:
-            host["name"] = re.sub(r"\W+", "", host["distro"])
-        status_line += "🗲%s " % host["name"]
-        print(status_line)
-        domain = hv.create_domain()
-        domain.context(context)
-        domain.name(host["name"])
-        domain.ssh_key_file(
-            configuration.get("ssh_key_file", "~/.ssh/id_rsa.pub")
-        )
-        domain.username(configuration.get("username", getpass.getuser()))
-        domain.vcpus(host.get("vcpus"))
-        domain.memory(host.get("memory", 768))
-        domain.add_root_disk(host["distro"])
-        domain.add_swap_disk(host.get("swap_size", 1))
-        domain.attachBridge(configuration["bridge"])
-        domain.start()
-        sys.stdout.write(CURSOR_UP_ONE)
-        sys.stdout.write(ERASE_LINE)
-    print(status_line)
-    print("Done! You can now follow the deployment. To get the live status:")
-    print("  vl status")
-    print("")
-    print("You can also access the serial console of the VM:")
-    print("  virsh console $vm_name")
+    if not os.path.isfile(virt_lightning_yaml_path):
+        print("Configuration file not found")
+        return
+
+    try:
+        with open(virt_lightning_yaml_path, "r") as fd:
+            host_definitions = yaml.load(fd)
+            hv = vl.LibvirtHypervisor(configuration)
+            print("Starting:")
+            status_line = ""
+            for host in host_definitions:
+                if "name" not in host:
+                    host["name"] = re.sub(r"\W+", "", host["distro"])
+                # Unfortunatly, i can't decode that symbol
+                # that symbol more well add to check encoding block
+                status_line += "🗲{hostname} ".format(
+                    hostname = host["name"]
+                )
+                print(status_line)
+                domain = hv.create_domain()
+                domain.context(context)
+                domain.name(host["name"])
+                domain.ssh_key_file(
+                    configuration.get("ssh_key_file", "~/.ssh/id_rsa.pub")
+                )
+                domain.username(configuration.get("username", getpass.getuser()))
+                domain.vcpus(host.get("vcpus"))
+                domain.memory(host.get("memory", 768))
+                domain.add_root_disk(host["distro"])
+                domain.add_swap_disk(host.get("swap_size", 1))
+                domain.attachBridge(configuration["bridge"])
+                domain.start()
+                sys.stdout.write(CURSOR_UP_ONE)
+                sys.stdout.write(ERASE_LINE)
+
+            print(status_line)
+
+            print("Done! You can now follow the deployment. To get the live status:")
+            print("  vl status")
+            print("")
+            print("You can also access the serial console of the VM:")
+            print("  virsh console $vm_name")
+    except IOError:
```

Comment:
I would prefer to isolate the try/catch a bit more. I don't think is make sense to wrap the whole domain creation.